File: CppUnit/TestSuite.cpp
#include <string>
#include "TestSuite.hpp"
#include <iostream>
#include <typeinfo>

using namespace std;
using namespace unit;

void TestCase::setUp(){

}

void WasRun::run(){
  this->setUp();
  cout << typeid(*this).name();
}

void TestCaseTest::testSetUp(){
  WasRun wr= WasRun("testMethod");
  cout << wr.wasRun << endl;
  wr.run();
  cout << wr.wasRun << endl;
}

WasRun::WasRun(string name){
	
}

void WasRun::setUp(){
  this->wasRun = false;
}

void WasRun::testMethod(){
  this->wasRun = true;
}

File: JavaUnit/TestSuite.java
package JavaUnit; 
import java.lang.reflect.Method;

class TestSuite {

  public static void main(String[] args) throws Exception{
	System.out.println("Running test suite...");
	WasRun test= new WasRun("testMethod");
    assert (test.wasRun == false) : "bad before run";
	test.run();
	assert (test.wasRun == true) : "bad after run";
  }
}

class TestCase{
  protected String name;
  protected boolean wasRun;

  TestCase(String name){
    this.name= name;
  }

  public void run() throws Exception{
    Method method = WasRun.class.getMethod(this.name);
    method.invoke(this);
  }
}

//class TestCaseTest extends TestCase{
//  TestCaseTest(String name){
//    super(name);
//  }
//  public void testRunning(){
//    WasRun test= new WasRun("testMethod");
//    assert (test.wasRun == false) : "bad before run";
//	try{
//	  super.run();
//    } catch(Exception e){
//	  System.out.println("Bad run");
//	}	
//	assert (test.wasRun == true) : "bad run";
//  }
//}

class WasRun extends TestCase {

  WasRun(String name){
	super(name);
    super.wasRun= false;
  }

  public void testMethod(){
    this.wasRun= true;
  }
}



